src/utils.ts

```diff
@@ -4,13 +4,39 @@ import { ChartModal } from "./ChartModal";
 import { DECIMALS } from "./const";
 import DataAnalysisPlugin from "./main";
 import { StatsModal } from "./StatsModal";
+import { DataType } from "./interfaces";
+
+export const createDataMapByFieldFromData = (data: { [field: string]: DataType }[], fields: string[]): {[field: string]: DataType[]} => {
+	let result: {[field: string]: DataType[]} = {};
+	data.forEach((page) => {
+		fields.forEach((field: string) => {
+			let fieldResult = result[field] || []
+			const value = page[field];
+			if (value) fieldResult.push(...makeArr(value));
+			result[field] = fieldResult
+		})
+	});
+	return result;
+}
+
+export const buildAllPairs = (items: string[]): string[][] => {
+	let results = [];
+
+	for (let outerIndex = 0; outerIndex < items.length; outerIndex++) {
+		let first = items[outerIndex]
+		for (let innerIndex=outerIndex+1; innerIndex < items.length; innerIndex++) {
+			results.push([first, items[innerIndex]])
+		}
+	}
+	return results;
+}
 
 export const splitAndTrim = (fields: string): string[] => {
 	if (fields === "") return [];
 	else return fields.split(",").map((str) => str.trim());
 };
 
-export function makeArr<T>(input: T | T[]): T[] {
+export function makeArr<T>(input: T | T[]): any[] {
```

Comment:
For some reason having this be a T was a type error in VS Code and I couldn't run the tests because of it? Let me know if you know of a better resolution here, I didn't have too much time to play with it yet.

Reply:
Yeah, I don't know why it doesn't allow the generic to be used... this is ok for now